packages/epo-react-lib/src/layout/SlideoutMenu/MenuItemRadio/MenuItemRadio.tsx

```diff
@@ -0,0 +1,78 @@
+import { FunctionComponent, useContext, useRef, useEffect } from "react";
+import MenuContext from "@/contexts/Menu";
+import { StyledMenuItemRadio, StyledMenuItemWrapper } from "./styles";
+import { useKeyDownEvent } from "@/hooks/listeners";
+
+interface MenuItemRadioProps {
+  text: string;
+  isChecked: boolean;
+  onCheckCallback: (close?: boolean) => void;
+}
+
+const MenuItemRadio: FunctionComponent<MenuItemRadioProps> = ({
+  text,
+  isChecked,
+  onCheckCallback,
+}) => {
+  const menuContext = useContext(MenuContext);
+  const menuItemRef = useRef<HTMLDivElement>(null);
+
+  if (!menuContext) {
+    throw new Error("Menu item must be used within a Menu Context");
+  }
+
+  const { menuItems, currentIndex } = menuContext;
+
+  useEffect(() => {
+    const menuItemNode = menuItemRef.current;
+
+    if (menuItemNode) {
+      menuItems && menuItems.add(menuItemNode);
+    }
+
+    return () => {
+      menuItems && menuItemNode && menuItems.delete(menuItemNode);
+    };
+  }, [menuItems, currentIndex]);
+
+  const isActive =
+    menuItemRef.current &&
+    [...menuItems].indexOf(menuItemRef.current) === currentIndex;
+
+  function handleKeyDown(event: KeyboardEvent) {
```

Comment:
`menuItemRadio` is basically the same as `menuItem` but with some keydown events and is styled as a radio button using `aria-checked`